lua/nvim-tree/view.lua

```diff
@@ -66,17 +66,52 @@ M.View = {
   }
 }
 
+---Find a rogue NvimTree buffer that might have been spawned by i.e. a session.
+---@return integer|nil
+local function find_rogue_buffer()
+  for i = 1, vim.fn.bufnr("$"), 1 do
+    if vim.fn.bufname(i) == "NvimTree" then
+      return i
+    end
+  end
+  return nil
+end
+
+---Check if the tree buffer is valid and loaded.
+---@return boolean
+local function is_buf_valid()
+  return a.nvim_buf_is_valid(M.View.bufnr) and a.nvim_buf_is_loaded(M.View.bufnr)
+end
+
+---Find pre-existing NvimTree buffer, delete its windows then wipe it.
+---@private
+function M._wipe_rogue_buffer()
+  local bn = find_rogue_buffer()
+  if bn then
+    local win_ids = a.nvim_eval("win_findbuf(" .. bn .. ")")
```

Comment:
`vim.fn.win_findbuf`

Reply:
Wow, I was just looking for this, and it didn't show up on tab completion for me so I didn't realize it was implemented.

Reply:
most of the time, you can just try it `:lua vim.fn.win_findbuf()` and it will crash saying either the function doesn't exist or the function misses parameters :)